python/ray/dag/tests/experimental/test_accelerated_dag.py

```diff
@@ -461,6 +461,75 @@ def test_chain_dag(ray_start_regular, num_actors):
     compiled_dag.teardown()
 
 
+def test_execution_timeout(ray_start_regular):
+    a = Actor.remote(0)
+    with InputNode() as inp:
+        dag = a.inc.bind(inp)
+
+    compiled_dag = dag.experimental_compile(execution_timeout=3)
+    refs = []
+    timed_out = False
+    for i in range(5):
+        try:
+            ref = compiled_dag.execute(1)
+            # Hold the refs to avoid get() being called on the ref
+            # when it goes out of scope
+            refs.append(ref)
+        except RayChannelTimeoutError:
+            # The first 3 tasks should complete, and the 4th one
+            # should block then time out because the max possible
+            # concurrent executions for the DAG is 3.
```

Comment:
Could you expand this comment to explain why three concurrent executions are possible, but not 4? Providing a visualization of the nodes and channels could be helpful.

Reply:
Sure.